pkg/operator/controller/ingress_status_test.go

```diff
@@ -6,13 +6,200 @@ import (
+	indexers := map[string]toolscache.IndexFunc{}
+	for name := range funcs {
+		fn := funcs[name]
+		indexers[name] = func(obj interface{}) ([]string, error) {
+			return fn(obj.(runtime.Object)), nil
+		}
+	}
+	return indexers
+}
+
+func TestComputeLoadBalancerStatus(t *testing.T) {
+	tests := []struct {
+		desc       string
+		controller string
+		strat      operatorv1.EndpointPublishingStrategyType
+		services   []*corev1.Service
+		expect     []operatorv1.OperatorCondition
+	}{
+		{
+			desc:       "provisioned/ready",
+			controller: "default",
+			strat:      operatorv1.LoadBalancerServiceStrategyType,
+			services: []*corev1.Service{
+				provisionedLBservice("secondary"),
+				provisionedLBservice("default"),
+				clusterIPservice("default"),
+			},
+			expect: []operatorv1.OperatorCondition{
+				cond(operatorv1.LoadBalancerManagedIngressConditionType, operatorv1.ConditionTrue, "HasLoadBalancerEndpointPublishingStrategy"),
+				cond(operatorv1.LoadBalancerReadyIngressConditionType, operatorv1.ConditionTrue, "LoadBalancerProvisioned"),
+			},
+		},
+		{
+			desc:       "pending/unready",
+			controller: "default",
+			strat:      operatorv1.LoadBalancerServiceStrategyType,
+			services: []*corev1.Service{
+				provisionedLBservice("secondary"),
+				pendingLBService("default"),
+				clusterIPservice("default"),
+			},
+			expect: []operatorv1.OperatorCondition{
+				cond(operatorv1.LoadBalancerManagedIngressConditionType, operatorv1.ConditionTrue, "HasLoadBalancerEndpointPublishingStrategy"),
+				cond(operatorv1.LoadBalancerReadyIngressConditionType, operatorv1.ConditionFalse, "LoadBalancerPending"),
+			},
+		},
+		{
+			desc:       "unmanaged",
+			controller: "default",
+			strat:      operatorv1.HostNetworkStrategyType,
+			services: []*corev1.Service{
+				provisionedLBservice("secondary"),
+				clusterIPservice("default"),
+			},
+			expect: []operatorv1.OperatorCondition{
+				cond(operatorv1.LoadBalancerManagedIngressConditionType, operatorv1.ConditionFalse, "UnsupportedEndpointPublishingStrategy"),
+			},
+		},
+		{
+			desc:       "missing",
+			controller: "default",
+			strat:      operatorv1.LoadBalancerServiceStrategyType,
+			services: []*corev1.Service{
+				provisionedLBservice("secondary"),
+				clusterIPservice("default"),
+			},
+			expect: []operatorv1.OperatorCondition{
+				cond(operatorv1.LoadBalancerManagedIngressConditionType, operatorv1.ConditionTrue, "HasLoadBalancerEndpointPublishingStrategy"),
+				cond(operatorv1.LoadBalancerReadyIngressConditionType, operatorv1.ConditionFalse, "LoadBalancerNotFound"),
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Logf("evaluating test %s", test.desc)
+		controllers := toolscache.NewIndexer(toolscache.MetaNamespaceKeyFunc, makeIndexers(controllerIndexers))
+		services := toolscache.NewIndexer(toolscache.MetaNamespaceKeyFunc, makeIndexers(serviceIndexers))
+		status := &ingressStatusCache{
+			contains: func(kind runtime.Object, name, value string) bool {
```

Comment:
Can probably find a way to make this more generic, but good enough for now. Can't use the `cache.Cache` interfaces in this context without setting up an apiserver.